src/mongo/db/auth/security_file.cpp

```diff
@@ -53,11 +53,35 @@ StatusWith<std::string> readSecurityFile(const std::string& filename) {
     }
 
 #if !defined(_WIN32)
-    // check permissions: must be X00, where X is >= 4
-    if ((stats.st_mode & (S_IRWXG | S_IRWXO)) != 0) {
+    if (stats.st_uid == 0)
+    {
+      /* In case the owner is root then permission of the key file
+       * can be a bit more open than for the non root users. The owner
+       * write and group read are also permissible values for the
+       * file permission.
+       * -rw--r---- 640, read and write owner of the file and read
+       * bit of group.
+       * These remaining bit should not be set for key file.
+       * S_IXUSR -- Owner Execute.
```

Comment:
Not sure we should prohibit "owner execute" bit. Upstream code does not do this though claims "X is >= 4". Do you think this is a bug in upstream?

Reply:
I really don't know what is the purpose of execute's bit of keyfile. Even it is not a bug upstream but it does not make sense to me, to have a execute permission on a keyfile.  Please advice otherwise

Reply:
I checked with ssh/opessl they are also alowing execute bit, so I am will update the patch.